src/StatsigOnPrem.ts

```diff
@@ -92,16 +92,16 @@ export default class StatsigOnPrem implements StatsigInterface {
     sdkKey: string,
     options?: ConfigSpecsOptions
   ): Promise<ConfigSpecs> {
-    const registeredKeys = await this.store.getRegisteredSDKKeys();
-    if (!registeredKeys.has(sdkKey)) {
-      console.warn("Attempting to use a non-registered key");
-    }
-
     const cachedSpecs = await this.cache.getSpecs(sdkKey, options);
     if (cachedSpecs) {
```

Comment:
Moved the check for cached specs up since the cache should be invalidated if the sdkKey has been removed.